### PR TITLE
updates terraform azurerm provider

### PR DESCRIPTION
**Pull request by mikepw**
Updates the `azurerm` provider so we can use the latest released version of azure rm resources, benefit from bug fixes and improvements.

### CODE REVIEW

File: terraform/modules/k8s/main.tf [hunk 1]
```diff
@@ -8,11 +8,11 @@ resource "azurerm_resource_group" "k8s" {
...
-  name                 = "${var.name}-${var.environment}-k8s"
-  location             = azurerm_resource_group.k8s.location
-  resource_group_name  = azurerm_resource_group.k8s.name
-  dns_prefix           = var.k8s_dns_prefix
-  private_link_enabled = var.private_link_enabled
+  name                    = "${var.name}-${var.environment}-k8s"
+  location                = azurerm_resource_group.k8s.location
+  resource_group_name     = azurerm_resource_group.k8s.name
+  dns_prefix              = var.k8s_dns_prefix
+  private_cluster_enabled = var.private_link_enabled
```
Comment (dandds) [inline terraform/modules/k8s/main.tf:15]: Should we change this variable name to match?

Comment (mikepw) [reply terraform/modules/k8s/main.tf]: good call. fixed

File: terraform/providers/pwdev/provider.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 provider "azurerm" {
-  version         = "=2.0.0"
+  version         = "=2.10.0"
```
Comment (jesse-dds) [inline terraform/providers/pwdev/provider.tf:2]: Is this a typo? Should this have gone from "2.0.0" to "2.1.0" or did it really move up 10 versions?

Comment (mikepw) [reply terraform/providers/pwdev/provider.tf:2]: not a typo :)
https://github.com/terraform-providers/terraform-provider-azurerm/blob/v2.10.0/CHANGELOG.md

**Comment by dandds** [comment]
@mikepw I think this just needs to be rebased on staging and `script/format` to merge.

**Pull request merged**